OptimizelySDK/Customization/DefaultURLSessionDelegate.swift

```diff
@@ -0,0 +1,58 @@
+
+//
+/****************************************************************************
+* Copyright 2019, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import Foundation
+
+// delegate used for URLSessions, methods called during sendEvent
+open class DefaultURLSessionDelegate: NSObject, URLSessionDelegate, URLSessionDataDelegate {
+    
+    var event: EventForDispatch
```

Comment:
Is EventForDispatch the same thing as EventDispatcher? Because I believe I am holding an EventForDispatch, which would be removed once the event is sent/removed via sendEvent()

Reply:
yeah, the urlsession has a retain on the delegate.  so, there is no retain cycle here.